crates/database/src/quotes.rs

```diff
@@ -20,6 +20,7 @@ pub struct Quote {
     pub sell_token_price: f64,
     pub order_kind: OrderKind,
     pub expiration_timestamp: DateTime<Utc>,
+    pub expiration_for_api_call_timestamp: Option<DateTime<Utc>>,
```

Comment:
I believe it would be slightly cleaner if this field would be required instead of optional. This way normal orders could just have the same value for both timestamps and we would not have to know that this timestamp only differs for specific order types (i.e. no `match`ing on the signature type, no additional parameter for `get_quote_and_check_fee()`).
If you go down that road the DB migration script would have to populate the new field for existing quotes appropriately.

Reply:
Also at some point we said we wouldn't want to bake support of ETH flow orders too much into the protocol (to have a better abstraction allowing others to build stuff on top of us). Is that still the case if we add this new field directly in the standard quote data?

Reply:
>I believe it would be slightly cleaner if this field would be required instead of optional.
I was making it optional as for pre-sign orders there is no expiration_for_api_call quotes, as pre-sign orders can only be placed on-chain.

> Is that still the case if we add this new field directly in the standard quote data?
I think yes. This feature comes from the fact that eip1271 orders can be placed on-chain and off-chain and we want different quotes expirations. It is not yet linked directly to Eth-flow orders.

Reply:
>If you go down that road the DB migration script would have to populate the new field for existing quotes appropriately.

Yeah, probably, even with the optional field, it would be a little bit cleaner.

Reply:
> I was making it optional as for pre-sign orders there is no expiration_for_api_call quotes, as pre-sign orders can only be placed on-chain.

My assumption is that `expiration_for_api_call` is always less or equal to `expiration_timestamp`. The reasoning is that if the order is already expired for good it wouldn't make sense to still be able to use it in the context of API calls.
By that logic if pre-sign orders already have the normal `expiration_timestamp` wouldn't that mean that the upper limit of `expiration_for_api_call` would have to be the same value? 
I'm not very familiar with that order type so it might be a stupid question or I didn't fully get the meaning of the 2 timestamps or my assumption is wrong. 🙈

Reply:
I think we want two different expiration times:
- A shorter `expiration_for_api_call` if you `POST api/v1/orders { ... signingScheme: presign ... }`
- A longer `expiration` if you execute a transaction that emits an **on-chain** `Order { ... signingScheme: presign ... }` order.